src/iop/finalscale.c

```diff
@@ -57,6 +58,12 @@ dt_iop_colorspace_type_t default_colorspace(dt_iop_module_t *self,
   return IOP_CS_RGB;
 }
 
+static inline gboolean _gui_fullpipe(dt_dev_pixelpipe_iop_t *piece)
+{
+  return piece->pipe->type & (DT_DEV_PIXELPIPE_FULL | DT_DEV_PIXELPIPE_PREVIEW2 | DT_DEV_PIXELPIPE_IMAGE)
```

Comment:
I suppose that the issue is there. A snapshot is DT_DEV_PIXELPIPE_PREVIEW IIRC, so not enabled by this check.

So maybe we need to relax this check and to launch a recomputing if the snapshot.

Reply:
BTW, we have the same issue with clone:
- Enter darkroom
- Clone an image
- Activate the "toggle export view scaling"
- Press and hold the mouse button over the other clone, see the displacement

Reply:
> I suppose that the issue is there. A snapshot is `DT_DEV_PIXELPIPE_PREVIEW` IIRC, so not enabled by this check.
> 
> So maybe we need to relax this check and to launch a recomputing if the snapshot.

Hum... maybe not the issue. It may well be because the full image processing state is not part of the context.

See `dt_view_get_view_context`.

Reply:
Maybe we need to keep the full-state read from the button somehow in snapshot status. Don't know if we can use the pipe. Using will investigate...

Reply:
Last comment crossed yours.

Reply:
I have tested adding the status into the context in `dt_view_get_view_context` and all works fine.

Reply:
Ok, I'll do that and will push a new PR for review and more testing.